src/Sparrow/Utils/NativeMemory.cs

```diff
@@ -30,7 +30,7 @@ public static void NotifyCurrentThreadAboutToClose()
 
         public static IEnumerable<ThreadStats> AllThreadStats => ThreadAllocations.Values.Where(x => x != null);
 
-        public static ConcurrentDictionary<string, ConcurrentDictionary<IntPtr, long>> FileMapping = new ConcurrentDictionary<string, ConcurrentDictionary<IntPtr, long>>();
+        public static ConcurrentDictionary<string, Lazy<FileMappingInfo>> FileMapping = new ConcurrentDictionary<string, Lazy<FileMappingInfo>>();
```

Comment:
why are you using lazy here?

Reply:
There could be multiple mappings for the same file. (increasing the file size or running in 32-bit)